src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -377,6 +378,66 @@ def list_key(cmd,
         raise CLIError(str(exception))
 
 
+def restore_key(cmd,
+                datetime,
+                key=None,
+                name=None,
+                label=None,
+                no_deletes=False,
+                connection_string=None,
+                yes=False):
+    current_retry = 0
+    max_retry = 3
+    retry_interval = 1
+
+    connection_string = resolve_connection_string(cmd, name, connection_string)
+    azconfig_client = AzconfigClient(connection_string)
+
+    query_option_then = QueryKeyValueCollectionOptions(key_filter=key,
+                                                       label_filter=QueryKeyValueCollectionOptions.empty_label if label is not None and not label else label,
+                                                       query_datetime=datetime)
+    query_option_now = QueryKeyValueCollectionOptions(key_filter=key,
+                                                      label_filter=QueryKeyValueCollectionOptions.empty_label if label is not None and not label else label)
+
+    try:
+        restore_keyvalues = azconfig_client.get_keyvalues(query_option_then)
+        current_keyvalues = azconfig_client.get_keyvalues(query_option_now)
+        kvs_to_restore, kvs_to_modify, kvs_to_delete = __compare_kvs_for_restore(restore_keyvalues, current_keyvalues, no_deletes)
+
+        if not yes:
+            need_change = __print_restore_preview(kvs_to_restore, kvs_to_modify, kvs_to_delete)
+            if need_change is False:
+                return
+
+        for kv in chain(kvs_to_restore, kvs_to_modify):
+            success = None
+            while success is None and current_retry <= max_retry:
+                try:
```

Comment:
what should we do if it still fails after retry?

Reply:
If it fails, we give up after 3. We don't keep trying to restore the others, though. Best to leave it there, log the relevant information, and let the user know.